homeassistant/components/lyric/climate.py

```diff
@@ -217,23 +217,26 @@ def target_temperature(self) -> float | None:
             if self.hvac_mode == HVAC_MODE_COOL:
                 return device.changeableValues.coolSetpoint
             return device.changeableValues.heatSetpoint
-        return None
 
     @property
     def target_temperature_high(self) -> float | None:
         """Return the highbound target temperature we try to reach."""
         device = self.device
-        if device.changeableValues.autoChangeoverActive:
+        if (
+            device.changeableValues.autoChangeoverActive
+            and HVAC_MODES[device.changeableValues.mode] != HVAC_MODE_OFF
+        ):
             return device.changeableValues.coolSetpoint
-        return None
```

Comment:
We need to return a value explicitly everywhere if we return a value anywhere in a function. Please keep this.

Reply:
Roger that. Will change back. Thanks for the guidance.